### DISCUSSION TITLE
Change progress color

### DISCUSSION THREAD

**Discussion by leonardoRocchini** [shadcn-ui/ui#1913] [Q&A]
Hello, i have question there is a way to change te color of the progress value from className ?
In the following code, I can change the background color but not the value color.
`<Progress  value={value} className='bg-green-700'/>`

**Comment by Alexjoshua14** [comment]
It seems like changing the color of the progress indicator directly through a className is not supported because the className you're applying is associated with the `ProgressPrimitive.Root`. To customize the color of the value indicator, you could modify the styles within the `Progress` component itself.

Here’s a quick solution I put together using an additional property to set the color of the progress indicator. 

![code](https://github.com/shadcn-ui/ui/assets/59298565/cb9aefb9-9a6e-457b-b5cd-224e9cc3260b)

This enables Progress to be customized like:

![Progress-Example](https://github.com/shadcn-ui/ui/assets/59298565/797e62bc-18cb-4fe2-8a0f-17e119f4e8e0)

**Reply by leonardoRocchini** [reply]
Thanks. Nice solution

**Comment by inj-src** [comment]
you can access the progress bar div with 
`<Progress
                  className="bg-gray-200 [&>div]:bg-blue-500 [&>div]:rounded-full h-1.5"
               />`

**Reply by IykeMich** [reply]
thank you so much

**Comment by Edmond-XavierCollot** [comment]
Update `bg-primary` to `bg-current` on the Indicator to allow color inheritance:

```tsx
<ProgressPrimitive.Indicator
  data-slot="progress-indicator"
  className="bg-current h-full w-full flex-1 transition-all"
  style={{ transform: `translateX(-${100 - (value || 0)}%)` }}
/>
```

And set the color with `text`:

```tsx
<Progress value={50} className="text-blue-500" />
```

**Comment by ryandotelliott** [comment]
I think the intended selector here, assuming you don't want to change the primitives themselves, would be to use `*:data-[slot=progress-indicator]:bg-green-700`. It's a bit more explicit than the `[&>div]`approach.
